src/main/java/org/jenkinsci/plugins/workflow/steps/CatchErrorStep.java

```diff
@@ -114,6 +116,25 @@ public void setCatchInterruptions(boolean catchInterruptions) {
         this.catchInterruptions = catchInterruptions;
     }
 
+    private void readObject(ObjectInputStream ois) throws ClassNotFoundException, IOException {
+        ObjectInputStream.GetField fields = ois.readFields();
+        message = (String) fields.get("message", null);
+        catchInterruptions = fields.get("catchInterruptions", true);
+        // Previously, the types of buildResult and stageResult were Result rather than String, so we handle either type.
```

Comment:
This is a bit of a mess. We could mark the plugin as incompatible and break any Pipeline that runs across the upgrade using `catchError`, but it seems better to try to maintain compatibility.